demosplan/DemosPlanCoreBundle/DoctrineMigrations/2025/04/Version20250415155558.php

```diff
@@ -0,0 +1,45 @@
+<?php declare(strict_types = 1);
+
+namespace Application\Migrations;
+
+use Doctrine\DBAL\Exception;
+use Doctrine\DBAL\Platforms\MySQLPlatform;
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\Migrations\AbstractMigration;
+
+class Version20250415155558 extends AbstractMigration
+{
+    public function getDescription(): string
+    {
+        return 'refs DPLAN-15338: Create custom field configuration table ';
+    }
+
+    /**
+     * @throws Exception
+     */
+    public function up(Schema $schema): void
+    {
+        $this->abortIfNotMysql();
+        $this->addSql('CREATE TABLE custom_field_configuration (id CHAR(36) NOT NULL, source_entity_id VARCHAR(36) NOT NULL, source_entity_class VARCHAR(255) NOT NULL, target_entity_class VARCHAR(255) NOT NULL, configuration JSON DEFAULT NULL COMMENT \'(DC2Type:dplan.custom_field_configuration)\', create_date DATETIME NOT NULL, modify_date DATETIME NOT NULL, PRIMARY KEY(id)) DEFAULT CHARACTER SET UTF8 COLLATE `UTF8_unicode_ci` ENGINE = InnoDB');
+    }
```

Comment:
minor format to improve readability

```suggestion
     public function up(Schema $schema): void
    {
        $this->abortIfNotMysql();
        $this->addSql('
            CREATE TABLE custom_field_configuration (
                id CHAR(36) NOT NULL,
                source_entity_id VARCHAR(36) NOT NULL,
                source_entity_class VARCHAR(255) NOT NULL,
                target_entity_class VARCHAR(255) NOT NULL,
                configuration JSON DEFAULT NULL COMMENT \'(DC2Type:dplan.custom_field_configuration)\',
                create_date DATETIME NOT NULL,
                modify_date DATETIME NOT NULL,
                PRIMARY KEY(id)
            )
            DEFAULT CHARACTER SET UTF8 COLLATE `UTF8_unicode_ci` ENGINE = InnoDB
        ');
    }
```